torch/onnx/_internal/fx/fx_symbolic_graph_extractor.py

```diff
@@ -227,11 +220,15 @@ def generate_fx(
         )
         graph_module = replace_get_attr_with_placeholder_pass.run()
         replaced_attrs = replace_get_attr_with_placeholder_pass.replaced_attrs
+        append_extra_input_step = io_adapter.AppendPositionalsIntoArgsStep(
```

Comment:
@BowenBao we can use this example as a requirement for the pass + IO adapter design.

A pass could receive input/output adapter and append stuff as needed, for example
Another possibility is simply to not allow passes to mutate model input/output

Reply:
Agree. And it can be enforced that the pass without new input/output adapt step cannot mutate model input/output.

Reply:
Ok, I think the passes API should be the next logical step in the design, to streamline those use cases.
Hopefully https://github.com/pytorch/pytorch/pull/100017 will be merged soon enough to eliminate this tracer